### PR TITLE
fix attribute resolver on promoted properties

### LINKED ISSUE #336: Attribute used in an CPP property is disallowed in attribute resolving callstack of command handlers
**Ecotone version(s) affected**: 1.224.0

**Description**  

The following issue occurs on warmup, when using an Attribute which should be allowed in this place in CPP

`Uncaught Error: Attribute "Symfony\Component\DependencyInjection\Attribute\AutowireLocator" cannot target property (allowed targets: parameter)`

This attribute(Symfony\Component\DependencyInjection\Attribute\AutowireLocator) for example is allowed by symfony to be used for properties in a CPP definition

```php
 public function __construct(
        #[AutowireLocator(FooLocatedService::class)]
        private ContainerInterface $foos,
    ) {
    }
```

So i think this should be allowed by ecotone too

**How to reproduce**  
I set an example symfony application with ecotone integration where this can reproduced at cache warmup
https://github.com/tandev/ecotone-attribute-resolving/tree/main

### PR DESCRIPTION
**Pull request by jlabedo**
## Description

Allow attributes targeting parameters to be used on php>8 promoted properties

Fixes #336 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/Ecotone/src/AnnotationFinder/AnnotationResolver/AttributeResolver.php [hunk 1]
```diff
@@ -81,7 +82,11 @@ public function getAnnotationsForProperty(string $className, string $propertyNam
                         return $carry;
                     }
 
-                    $carry[] = $attribute->newInstance();
+                    try {
+                        $carry[] = $attribute->newInstance();
+                    } catch (Error) {
+                        // Do nothing: it is an attribute targeting a parameter promoted to a property
```
Comment (dgafka) [inline packages/Ecotone/src/AnnotationFinder/AnnotationResolver/AttributeResolver.php:88]: That's assumption that this is the error :). We may be swallowing others errors too.
If we can't check by exception type, can we do an message type check e.g. string contains `cannot target property`?

Comment (jlabedo) [reply packages/Ecotone/src/AnnotationFinder/AnnotationResolver/AttributeResolver.php:88]: Sure

File: packages/Ecotone/src/AnnotationFinder/AnnotationResolver/AttributeResolver.php [hunk 2]
```diff
@@ -84,8 +84,12 @@ public function getAnnotationsForProperty(string $className, string $propertyNam
 
                     try {
                         $carry[] = $attribute->newInstance();
-                    } catch (Error) {
-                        // Do nothing: it is an attribute targeting a parameter promoted to a property
+                    } catch (Error $e) {
+                        if (\preg_match("/Attribute \"(.*)\" cannot target property/", $e->getMessage())) {
+                            // Do nothing: it is an attribute targeting a parameter promoted to a property
+                        } else {
```
Comment (dgafka) [inline packages/Ecotone/src/AnnotationFinder/AnnotationResolver/AttributeResolver.php:90]: Can we also apply this to `TypeResolver#86`? It's other way around there and we ignore all exceptions.

Comment (jlabedo) [reply packages/Ecotone/src/AnnotationFinder/AnnotationResolver/AttributeResolver.php:90]: Ha, you found it ! I knew it was done elsewhere.

**Review by dgafka — APPROVED** [review]

**Pull request merged**